src/index.js

```diff
@@ -23,26 +24,28 @@ const uploadBundle = require("./upload");
  * @returns {Promise<{ handlers: string[], mainFile: string }>} list of handlers and path to entrypoint
  */
 async function assemble(src) {
-  const tmpDir = await fsPromises.mkdtemp("handlers-"); //make temp dir `handlers-abc123`
+  const tmpDir = await fsPromises.mkdtemp(path.join(os.tmpdir(), "handlers-")); //make temp dir `handlers-abc123`
   const handlers = [];
-  let imports = "";
-  let registration = "";
+  const imports = [];
+  const registration = [];
   const functions = await fsPromises.readdir(src, { withFileTypes: true });
+
   for (const func of functions) {
-    if (!func.isFile() || (!func.name.endsWith(".js") && !func.name.endsWith(".ts"))) {
+    const file = path.parse(func.name);
+
+    if (!func.isFile() || (file.ext !== ".js" && file.ext !== ".ts")) {
       continue;
     }
 
     const id = "func" + crypto.randomBytes(16).toString("hex");
-    const name = func.name.substr(0, func.name.length - 3); // remove extension //
-    imports += `import * as ${id} from "${path.resolve(src, func.name)}";\n`;
-    registration += `netlifyRegistry.set("${name}", ${id});\n`;
 
-    handlers.push(func.name);
+    imports.push(`import * as ${id} from "${path.resolve(src, file.name)}";`);
+    registration.push(`netlifyRegistry.set("${file.name}", ${id});`);
```

Comment:
Nice, this makes it super clear that imports and registration are taking in multiple files